component/extension.go

```diff
@@ -50,6 +51,9 @@ type ExtensionCreateSettings struct {
 
 	// BuildInfo can be used by components for informational purposes
 	BuildInfo BuildInfo
+
+	// Gates can be used to query feature gate status
+	Gates featuregate.Gates
```

Comment:
Haha, this is easy to solve, see the trick with "component.TelemetrySettings".

Reply:
/cc @codeboten what do you think?

Reply:
that seems like a good way to limit the number of places the change needs to be made, though it still grows that surface with something that feels like it might belong more in `Host`.  Not sure where else `Host` is used though or if putting it there would actually buy anything.